app/services/middleware_topology_service.rb

```diff
@@ -33,14 +36,28 @@ def entity_display_type(entity)
 
   def build_entity_data(entity)
     data = build_base_entity_data(entity)
-    data.merge!(:status => 'Unknown',
-                :display_kind => entity_display_type(entity))
-    data[:icon] = entity.decorate.try(:listicon_image)
+    data[:status] = 'Unknown'
+    data[:display_kind] = entity_display_type(entity)
+
+    # fonticon takes precedence before the listicon_image
+    # and both methods on decorator takes precedence before the global icons hash from the UiServiceMixin
+    data[:icon] = if !(icon = entity.decorate.try(:fonticon)).nil?
```

Comment:
why is the icon added here to every entity if it's already added once for each entity type in the icons variable that is passed to frontend?